ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs

```diff
@@ -909,8 +939,22 @@ translateLedgerStateMaryToAlonzoWrapper ::
        (ShelleyBlock (AlonzoEra c))
 translateLedgerStateMaryToAlonzoWrapper =
     RequireBoth $ \_cfgMary cfgAlonzo ->
-      TranslateLedgerState $ \_epochNo ->
-        unFlip . unComp . SL.translateEra' (getAlonzoTranslationContext cfgAlonzo) . Comp . Flip
+      TranslateLedgerState {
+          translateLedgerStateWith = \_epochNo ->
+                (flip withLedgerTables polyEmptyLedgerTables)
+              . unFlip
+              . unComp
+              . SL.translateEra' (getAlonzoTranslationContext cfgAlonzo)
+              . Comp
+              . Flip
+        , translateLedgerTablesWith =
+            \ShelleyLedgerTables { shelleyUTxOTable = ApplyValuesMK (HD.UtxoValues vals) } ->
+             ShelleyLedgerTables { shelleyUTxOTable = ApplyValuesMK
+                                                    . HD.UtxoValues
+                                                    . fmap Alonzo.translateTxOut
```

Comment:
Perhaps `translateTxOut` should be performed through `SL.translateEra' ()` as it seems an instance could be defined for that one on the ledger side. This would continue with the same pattern as eras above